R/annotate.R

```diff
@@ -0,0 +1,53 @@
+# Annotate Graphs (should this go into gGtools.R?)
+#
+#' Produce annotated gene graph where vertices have scores
+#' attributed to proper genes.
+#'
+#' What needs to happen for this to work:
+#' - the gene heat scores should be generated
+#'
+#' @param gH List of genes and their associated score (mutational frequncy of
+#'           mutSigCV value)
+#' @param gG gene graph
+#' @param silent Controls whether output to console should be suppressed. FALSE
+#'   by default.
+#' @param writeLog Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#' @return gG object with annotated vertices
+#'
+
+annotateGraph <- function(gH, gG, silent = FALSE, writeLog = TRUE) {
+
+    # check for params - that they are not missing
+
+    # Since gH format is a hash, should:
+    # 1. create a dataframe of all of the vertices (igraph::vertex.attributes(gG))
+    # 2. annotate the vertices at once using hash, create a list of them ordered
+    # 3. attach the corresponding vertex annotations to gG using igraph::set_vertex_attr(gG, "heat", <list of heatValues>)
```

Comment:
I suspect that it will be far more efficient to pass the entire set of vertex-attributes into the graph in one operation than iterating over the vertices one by one.